### PR TITLE
Add namespaced rbac mode

### PR DESCRIPTION
**Pull request by yaron2**
Closes https://github.com/dapr/dapr/issues/5787.

This PR enables two settings that both individually and combined allow for stricter RBAC permissions and bring us closer to being able to deploy multiple control plane installations per cluster.

### CODE REVIEW

File: pkg/operator/watchdog.go [hunk 1]
```diff
@@ -196,7 +196,7 @@ func (dw *DaprWatchdog) listPods(ctx context.Context) bool {
 		log.Debugf("Taking a pod restart token")
 		before := time.Now()
 		_ = dw.restartLimiter.Take()
-		log.Debugf("Resumed after pausing for %v", time.Now().Sub(before))
+		log.Debugf("Resumed after pausing for %v", time.Since(before))
```
Comment (yaron2) [inline pkg/operator/watchdog.go:199]: This change is unrelated to the PR and follows linter recommendations.

File: charts/dapr/README.md [hunk 2]
```diff
@@ -99,6 +99,7 @@ The Helm chart has the follow configuration options that can be supplied:
 | `global.issuerFilenames.ca`               | Custom name of the file containing the root CA certificate inside the container    | `ca.crt`     |
 | `global.issuerFilenames.cert`             | Custom name of the file containing the leaf certificate inside the container       | `issuer.crt` |
 | `global.issuerFilenames.key`              | Custom name of the file containing the leaf certificate's key inside the container | `issuer.key` |
+| `global.rbac.namespaced`                  | Uses roles and roles bindings in all applicable places other than issuing tokens and listing service accounts for identity verification  | `false` |
```
Comment (ItalyPaleAle) [inline charts/dapr/README.md:102]: `roles bindings` -> `role bindings`

However not sure I fully understand what this means

File: charts/dapr/charts/dapr_rbac/templates/dashboard.yaml [hunk 3]
```diff
@@ -7,7 +7,11 @@ metadata:
     {{ $key }}: {{ tpl $value $ }}
     {{- end }}
 ---
+{{- if eq .Values.global.rbac.namespaced true }}
```
Comment (ItalyPaleAle) [inline charts/dapr/charts/dapr_rbac/templates/dashboard.yaml:10]: I think

```
{{- if .Values.global.rbac.namespaced }}
```

would be enough?

File: charts/dapr/charts/dapr_rbac/templates/injector.yaml [hunk 4]
```diff
@@ -19,9 +19,11 @@ rules:
   - apiGroups: [""]
     resources: ["serviceaccounts"]
     verbs: ["get", "list"]
+{{- if eq .Values.global.rbac.namespaced false }}
```
Comment (ItalyPaleAle) [inline charts/dapr/charts/dapr_rbac/templates/injector.yaml:22]:
```
{{- if not .Values.global.rbac.namespaced }}
```

File: cmd/operator/main.go [hunk 5]
```diff
@@ -114,6 +116,8 @@ func init() {
 	flag.IntVar(&maxPodRestartsPerMinute, "max-pod-restarts-per-minute", defaultMaxPodRestartsPerMinute, "Maximum number of pods in an invalid state that can be restarted per minute")
 	flag.BoolVar(&disableLeaderElection, "disable-leader-election", false, "Disable leader election for operator")
 
+	flag.StringVar(&watchNamespace, "watch-namespace", "", "Namespace to watch K8s resources")
```
Comment (ItalyPaleAle) [inline cmd/operator/main.go:119]: `Namespace to watch K8s resources in` or `Namespace to watch for K8s resources`

**Review by ItalyPaleAle — DISMISSED** [review]
LGTM just a few nits

File: charts/dapr/README.md (hunk 2 above)
Comment (mukundansundar) [reply charts/dapr/README.md:102]: Probably needs to be rephrased

File: cmd/operator/main.go (hunk 5 above)
Comment (mukundansundar) [reply cmd/operator/main.go:119]: Namespace to watch for Dapr annotated pod deployments

**Review by mukundansundar — COMMENTED** [review]
Qq does anything need to change in terms of dapr watch dog? 

Also can you create a PR in cli adding watch namespace as a flag for init -k?

**Comment by yaron2** [comment]
> Qq does anything need to change in terms of dapr watch dog?
> 
> Also can you create a PR in cli adding watch namespace as a flag for init -k?

The Watchdog becomes namespaced and doesn't need further updates.

**Review by mukundansundar — APPROVED** [review]
LGTM. I believe that we should add an E2E for this.

File: cmd/operator/main.go (hunk 5 above)
Comment (yaron2) [reply cmd/operator/main.go]: Rephrased

File: charts/dapr/charts/dapr_rbac/templates/injector.yaml (hunk 4 above)
Comment (yaron2) [reply charts/dapr/charts/dapr_rbac/templates/injector.yaml]: Changed

File: charts/dapr/charts/dapr_rbac/templates/dashboard.yaml (hunk 3 above)
Comment (yaron2) [reply charts/dapr/charts/dapr_rbac/templates/dashboard.yaml:10]: I've encountered Helm oddities before, not taking my chances :)

File: charts/dapr/README.md (hunk 2 above)
Comment (yaron2) [reply charts/dapr/README.md]: Rephrased

File: cmd/operator/main.go [hunk 6]
```diff
@@ -116,7 +116,7 @@ func init() {
 	flag.IntVar(&maxPodRestartsPerMinute, "max-pod-restarts-per-minute", defaultMaxPodRestartsPerMinute, "Maximum number of pods in an invalid state that can be restarted per minute")
 	flag.BoolVar(&disableLeaderElection, "disable-leader-election", false, "Disable leader election for operator")
 
-	flag.StringVar(&watchNamespace, "watch-namespace", "", "Namespace to watch K8s resources")
+	flag.StringVar(&watchNamespace, "watch-namespace", "", "Namespace to watch Dapr annotated resources in")
```
Comment (mukundansundar) [inline cmd/operator/main.go:119]: Nit: Namespace to watch `for` Dapr annotated resources in

**Review by mukundansundar — APPROVED** [review]
LGTM.
